src/utils/settings.ts

```diff
@@ -81,7 +81,12 @@ export const getSettings = async (): Promise<Settings> => {
 };
 
 export const saveSettings = async (settings: Settings): Promise<void> => {
-  return browser.storage.sync.set(settings);
+  // permissions.request must be the first await, not after the browser.storage.sync.set
+  // See https://stackoverflow.com/a/47729896/12601364
+  await browser.permissions.request({
```

Comment:
Should we skip requesting permissions if the api url is the default? Just in case it prompts for permissions when it already has them.

Reply:
It does not prompt for the default API URL on Firefox, and it does not prompt for even new API URLs on Chrome as it's implied by the content script permissions.